File: common/lua_common/file_io.lua
-- ----------------------------------------------------
--	file_io.lua
--
--					May/27/2011
-- ----------------------------------------------------
function file_to_str_proc (file_in)
	file, msg=io.open (file_in,"r")

	if (file) then
		str_in = file:read ("*a")
		file:close ()
	else
		print (msg)
	end

	return	str_in
end
-- ----------------------------------------------------
-- ----------------------------------------------------

File: common/erlang_common/file_io.erl
%% -----------------------------------------------------------
%%	file_io.erl
%%
%%					Sep/28/2011
%% -----------------------------------------------------------
-module (file_io).
-export ([file_to_str_proc/1]).
%% -----------------------------------------------------------
file_to_str_proc (File_in) ->
	Lout = [],
	{ok,S}=file:open(File_in,read),
	L1=read_single_line_proc (S,Lout),
	file:close (S),
	lists:concat (L1).
%% -----------------------------------------------------------
read_single_line_proc (S,Lout)->
	case io:get_line (S,'') of
	eof ->
		Lout;
	Line ->
		Lnew = Lout ++ [Line],
		read_single_line_proc (S,Lnew)
	end.
%% -----------------------------------------------------------
